ika/client_mock.go

```diff
@@ -6,26 +6,47 @@ import (
 	"github.com/block-vision/sui-go-sdk/models"
 	tmtypes "github.com/cometbft/cometbft/types"
 	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/mock"
 )
 
 // MockClient is a mock implementation of the Client interface for testing.
-type MockClient struct{}
+type MockClient struct {
+	mock.Mock
+}
 
-// NewMockClient creates a new mock Client instance.
+// NewMockClient creates a new mock Client instance with predefined ApproveAndSign behavior.
 func NewMockClient() *MockClient {
-	return &MockClient{}
+	mockClient := new(MockClient)
+	mockClient.On("ApproveAndSign", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
```

Comment:
Could we make approveAndSign constant? This is flexible when Ika updates their APIs.

Reply:
`ApproveAndSign` is a method defined for our client. The implementation of the method might change if we need to call different sui modules but whether the name change it up to us. I dont think we should move it to a constant